File: src/WWS/Mail/mail/src/main/kotlin/hu/bme/aut/mail/integrationEvent/event/InvoiceCreatedEvent.kt
package hu.bme.aut.mail.integrationEvent.event

import com.fasterxml.jackson.annotation.JsonProperty
import java.io.Serializable

data class InvoiceCreatedEvent(
    @JsonProperty("orderCode") val orderCode: String,
    @JsonProperty("totalPrice") val totalPrice: Double,
    @JsonProperty("created") val created: String,
    @JsonProperty("zip") val zip: String,
    @JsonProperty("city") val city: String,
    @JsonProperty("street") val street: String,
    @JsonProperty("countryCode") val countryCode: String,
    @JsonProperty("email") val email: String,
) : Serializable

File: src/WWS/Invoice/Web/IntegrationEvents/InvoiceCreatedEvent.cs
using System;
using System.Collections.Generic;
using System.Linq;
using System.Threading.Tasks;

namespace Web.IntegrationEvents
{
    public class InvoiceCreatedEvent : IInvoiceCreatedEvent
    {
        public double TotalPrice { get; set; }
        public string OrderCode { get; set; }
        public string Created { get; set; }
        public string Zip { get; set; }
        public string City { get; set; }
        public string Street { get; set; }
        public string CountryCode { get; set; }
        public string Email { get; set; }
    }

    public interface IInvoiceCreatedEvent
    {
        public double TotalPrice { get; set; }
        public string OrderCode { get; set; }
        public string Created { get; set; }
        public string Zip { get; set; }
        public string City { get; set; }
        public string Street { get; set; }
        public string CountryCode { get; set; }
        public string Email { get; set; }
    }
}
